custom/icds/location_reassignment/parser.py

```diff
@@ -56,14 +56,14 @@ def __init__(self, domain, workbook):
                 vii. there should be no new parent site code where not expected
                 viii. new location is created with different details for a reused parent
         2. Consolidated validations
-            a. if a location is archived, all its descendants should get archived too
```

Comment:
Ya that is one confusing this with the naming
Deprecate would not necessarily mean archival. There is one instance where deprecation will not archive the location. When we do an "Extract" operation on a location, we retain the old location and not archive it.
This operation was introduced after we had chosen the name deprecate which went well earlier because we were archiving for all operations back then, but now it has become confusing and we are looking for something apart from deprecate that wont cause this confusion. Open to suggestions for it.

Reply:
I see what you mean. I'm not sure there's a single word that would make sense here, so I don't think using deprecate makes things clearer. In most places "old" seems fine. For this line, the old code is true, right? All archived locations do have their descendants archived. That line just doesn't apply to extract operations, since they don't archive locations.

Reply:
Agreed, deprecate isn't doing well, need to come up with better name, currently leaning towards, `deprecate` -> `transition`.

You are right when catching that. I have updated comments to what is actually happening currently in the code, which is not really correct here. Currently we are checking for deprecated but should check for archived.
We should ideally check for only non-extract operations but as of now we are checking for all. Its a bug i have to fix and has not been caught in QA yet.

Reply:
i forgot, this was actually caught in UAT a day back
https://dimagi-dev.atlassian.net/browse/ICDS-1459